ext/pdo_firebird/firebird_statement.c

```diff
@@ -459,11 +464,19 @@ static int firebird_stmt_get_col(
 					break;
 				case SQL_FLOAT:
 					/* TODO: Why is this not returned as the native type? */
-					ZVAL_STR(result, zend_strpprintf(0, "%F", *(float*)var->sqldata));
+					{
+						char f_buf[256];
+						zend_gcvt((double) get_float_from_sqldata(var->sqldata), 8, '.', 'E', f_buf);
```

Comment:
Can't our custom ``%H`` format specifier with ``zend_spprintf_unchecked``/``[zend_strpprintf_unchecked](https://heap.space/s?defs=zend_strpprintf_unchecked&project=php-src)`` do the trick?

Reply:
Thanks, that makes sense! I fixed it.